### PR TITLE
feat: added all basic features of backend

### PR DESCRIPTION
**Pull request by sjsouvik**

### CODE REVIEW

**Review by vinaybajjuri58 — COMMENTED** [review]
Hey Souvik 
The code structure is pretty good nothing much changes to it 
But I suggest move paramHandlers to a seperate folder lto seperate it from controllers 
And remove commented line

File: shopwild-backend/controllers/cart.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      return res.json({ message: "Created cart for the user", cart });
+    } else {
+      let cartItem = cart.products.find(
+        (item) => item.product == cartUpdates.products[0].product
+      );
+      // cartItem = Cart.findOne({
+      //   "products.product": {
+      //     $eleMatch: { product: cartUpdates.products[0].product },
+      //   },
+      // });
```
Comment (Jaynil1611) [inline shopwild-backend/controllers/cart.js:50]: Remove commented code

File: shopwild-backend/controllers/cart.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    });
+  }
+};
+
+exports.getCart = async (req, res) => {
+  try {
+    const cart = await Cart.findOne({ user: req.user._id })
+      .populate("user")
+      .populate("products.product");
+    res.json({ cart });
```
Comment (Jaynil1611) [inline shopwild-backend/controllers/cart.js:22]: Add status code as `res.status()` while sending responses across the source code

File: shopwild-backend/controllers/cart.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+};
+
+exports.createUpdateCart = async (req, res) => {
+  try {
+    const cartUpdates = req.body;
+    let cart = await Cart.findOne({ user: req.user._id });
+
+    cartUpdates.user = req.user;
+
+    if (cart === null) {
```
Comment (Jaynil1611) [inline shopwild-backend/controllers/cart.js:38]: No need to compare with null,
```suggestion
    if (cart) {
```

File: shopwild-backend/controllers/wishlist.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    res.json({ wishlist });
+  } catch (error) {
+    res.status(404).json({
+      message: "Unable to get the cart for the user",
+      errorMessage: error.message,
+    });
+  }
+};
+
+exports.createUpdateWishlist = async (req, res) => {
```
Comment (Jaynil1611) [inline shopwild-backend/controllers/wishlist.js:18]: Rename function as `createOrUpdate`

File: shopwild-backend/index.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+const connectionString = process.env.DB_CONNECTION_STRING;
+mongoose
+  .connect(connectionString, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    useCreateIndex: true,
+    useFindAndModify: false,
+  })
+  .then(() => console.log("Connected to DB"))
```
Comment (Jaynil1611) [inline shopwild-backend/index.js:46]: use `async await` syntax

File: shopwild-backend/routes/user.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      const userUpdates = req.body;
+      user = extend(user, userUpdates);
+      user = await user.save();
+      res.json({ user });
+    } catch (error) {
+      res
+        .status(400)
+        .json({ message: "Error occured", errorMessage: error.message });
+    }
+  });
```
Comment (Jaynil1611) [inline shopwild-backend/routes/user.js:57]: Extract this as a function in the user controller to make the code consistent and readable

**Review by Jaynil1611 — CHANGES_REQUESTED** [review]
- Great work @sjsouvik 
- Folder structure and file naming conventions are consistent
- Here are my few suggestions:

**Pull request merged**